orchagent/fdborch.cpp

```diff
@@ -234,23 +432,20 @@ bool FdbOrch::getPort(const MacAddress& mac, uint16_t vlan, Port& port)
         return false;
     }
 
-    sai_fdb_entry_t entry;
-    entry.switch_id = gSwitchId;
-    memcpy(entry.mac_address, mac.getMac(), sizeof(sai_mac_t));
+    FdbEntry entry;
     entry.bv_id = port.m_vlan_info.vlan_oid;
-
-    sai_attribute_t attr;
-    attr.id = SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID;
-
-    sai_status_t status = sai_fdb_api->get_fdb_entry_attribute(&entry, 1, &attr);
-    if (status != SAI_STATUS_SUCCESS)
+    entry.mac = mac;
+    auto it= m_entries.find(entry);
+    if(it != m_entries.end())
+    {
+        bridge_port_id = it->second.bridge_port_id;
+    }
+    else
     {
-        SWSS_LOG_ERROR("Failed to get bridge port ID for FDB entry %s, rv:%d",
-            mac.to_string().c_str(), status);
         return false;
     }
 
-    if (!m_portsOrch->getPortByBridgePortId(attr.value.oid, port))
+    if (!m_portsOrch->getPortByBridgePortId(bridge_port_id, port))
     {
         SWSS_LOG_ERROR("Failed to get port by bridge port ID 0x%" PRIx64, attr.value.oid);
```

Comment:
`attr.value.oid` should change to `bridge_port_id` ?

Reply:
Yes definitely; I will correct it.
Thanks for spotting the error.

Reply:
Rectified in latest commit.